cpp/ql/test/library-tests/ir/ir/ir.cpp

```diff
@@ -2331,4 +2378,18 @@ namespace return_routine_type {
 
 }
 
+
+using size_t = decltype(sizeof(0));
+
+template<class T>
+struct remove_const { typedef T type; };
+
+template<class T>
+struct remove_const<const T> { typedef T type; };
+
+// `remove_const_t<T>` removes any `const` specifier from `T`
+template<class T>
+using remove_const_t = typename remove_const<T>::type;
+
+
```

Comment:
Not sure what the purpose of this is.

Reply:
`remove_const_t` is necessary for the declaration of the `non-const -> const conversion constructor` on `std::iterator`. I basically just copied it from [here](https://github.com/github/codeql/blob/a13391bda16dfb3ee2d04fdef0aeb61645cfbfbe/cpp/ql/test/include/iterator.h#L22). It may be the case that we can delete it since we don't use this conversion constructor in `ir.cpp`, though. I just didn't want the two to diverge _too_ much.

Reply:
But we now have two copies of this. One in the `std` namespace that was introduced higher up in the file, and one here at the very bottom of the file. I'm commenting here on the copy a the very bottom of the file.

Reply:
Oh! I totally missed that we had two of these! Let me fix up that!

Reply:
Fixed in 0be329dbdccc742fbd40317c4cf20c612faa37c6

Reply:
Now I think I know what happend:
1. I thought I had to introduce iterators to the `ir.cpp` file to get any test changes, so I added the code I just deleted + an `iterator` class template to observe test files.
2. I realized that we already had iterators in this file, but they were so barebone that our iterator detection didn't find them
3. I deleted the `iterator` class template from the file, but forgot to delete the `const` removal code